File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/mainmenus/MainMenuFirebaseFragment.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.mainmenus;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.pet.FirebaseFragment;
import com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.MainMenuViewModel;

/**
 * TitleScreenFragment class
 *
 * @author TritiumGamingStudios
 */
public abstract class MainMenuFirebaseFragment extends FirebaseFragment {

    protected MainMenuViewModel mainMenuViewModel;

    public MainMenuFirebaseFragment() {
        super();
    }

    public MainMenuFirebaseFragment(int layout) {
        super(layout);
    }

    protected void initViewModels() {
        super.initGlobalPreferencesViewModel();
    }

    protected void initMainMenuViewModel() {
        if (mainMenuViewModel == null) {
            try {
                mainMenuViewModel =
                        new ViewModelProvider(requireActivity()).get(MainMenuViewModel.class);
            } catch (IllegalStateException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    protected void backPressedHandler() {
        super.backPressedHandler();
    }

    @Override
    protected void saveStates() {}

}

File: app-view/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/activities/mainmenus/MainMenuFirebaseFragment.kt
package com.tritiumgaming.phasmophobiaevidencepicker.activities.mainmenus

import androidx.lifecycle.ViewModelProvider
import com.tritiumgaming.phasmophobiaevidencepicker.activities.pet.FirebaseFragment
import com.tritiumgaming.phasmophobiaevidencepicker.data.viewmodels.MainMenuViewModel

abstract class MainMenuFirebaseFragment : FirebaseFragment {

    protected var mainMenuViewModel: MainMenuViewModel? = null

    constructor() : super()

    constructor(layout: Int) : super(layout)

    override fun initViewModels() { super.initGlobalPreferencesViewModel() }

    protected fun initMainMenuViewModel() {
        if (mainMenuViewModel == null) {
            try { mainMenuViewModel =
                    ViewModelProvider(requireActivity())[MainMenuViewModel::class.java]
            } catch (e: IllegalStateException) { e.printStackTrace() }
        }
    }

    override fun saveStates() {}
}